node/node.go

```diff
@@ -841,10 +860,17 @@ func (node *AlgorandFullNode) IsArchival() bool {
 }
 
 // OnNewBlock implements the BlockListener interface so we're notified after each block is written to the ledger
+// The method is being called *after* the transaction pool received it's OnNewBlock call.
```

Comment:
I agree with that - the node should receive the callback and send it back to all the child services. I'd rather avoid doing this in the context of this PR, since it's not really related. ( and the call order is important, naturally )